nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java

```diff
@@ -286,4 +309,37 @@ private Type.ClassType typeWithPreservedAnnotations(ParameterizedTypeTree tree)
             type.getEnclosingType(), com.sun.tools.javac.util.List.from(newTypeArgs), type.tsym);
     return finalType;
   }
+
+  public void checkTypeParameterNullnessForAssignabilityForConditionalExpression(
+      ConditionalExpressionTree tree) {
+    if (!config.isJSpecifyMode()) {
+      return;
+    }
+    //  Tree lhsTree;
+    Type lhsType;
+    Tree truePartTree;
+    Tree falsePartTree;
+
+    lhsType = ASTHelpers.getType(tree);
+    truePartTree = tree.getTrueExpression();
+    falsePartTree = tree.getFalseExpression();
+    Type truePartType = getTreeType(truePartTree);
+    Type falsePartType = getTreeType(falsePartTree);
+    // handling diamond operator case for now
+    if (truePartType.getTypeArguments().isEmpty()) {
+      return;
+    }
+    if (falsePartType.getTypeArguments().isEmpty()) {
+      return;
+    }
+    if (truePartType != null
+        && falsePartType != null
+        && falsePartType instanceof Type.ClassType
+        && truePartType instanceof Type.ClassType
+        && lhsType != null
+        && lhsType instanceof Type.ClassType) {
+      compareNullabilityAnnotations((Type.ClassType) truePartType, (Type.ClassType) lhsType, tree);
+      compareNullabilityAnnotations((Type.ClassType) lhsType, (Type.ClassType) falsePartType, tree);
```

Comment:
can't you just do one comparison of `truePartType` and `falsePartType`?  why do you need to compare to `lhsType`?

Reply:
Actually nevermind this is to check that the types of `truePartType` and `falsePartType` are consistent with the context in which the conditional expression is used.  Let's handle this a bit differently and see if it works, as follows:

* Here, only compare `truePartType` and `falsePartType`
* In the new `getTreeType` method, if the tree is a conditional expression, rather then just getting its type, return the type of the true expr (or the false expr, doesn't matter).  This way we will get different error messages when a bad conditional expression is assigned, returned, passed as a parameter, etc.